app/customer.py

```diff
@@ -0,0 +1,50 @@
+import json
+import math
+from dataclasses import dataclass
+from app.car import fuel_price
+from app.shop import Shop
+
+
+@dataclass
+class Customer:
+    name: str
+    product_cart: dict
+    location: list
+    money: float
+    car: dict
+
+    def distance(self, shop: Shop) -> float:
+        pow_x_coordinate = pow(self.location[0] - shop.location[0], 2)
+        pow_y_coordinate = pow(self.location[1] - shop.location[1], 2)
+        return math.sqrt(pow_x_coordinate + pow_y_coordinate)
+
+    def product_value(self, shop: Shop) -> float:
+        return sum(
+            [
+                self.product_cart[product_name] * shop.products[product_name]
+                for product_name in self.product_cart
+            ]
+        )

```

Comment:
no need to create list
```suggestion
        return sum(
            self.product_cart[product_name] * shop.products[product_name]
            for product_name in self.product_cart
        )

```